### PR TITLE
save updates to resources to disk async (wip)

### PR DESCRIPTION
**Pull request by sent-hil**
TODOS:
* Get confirmation this is the right way to do this
* Write tests

---
This change is [Reviewable](https://reviewable.io/reviews/sfbrigade/sfsbook/147)

### CODE REVIEW

File: dba/index_and_backup.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	listOfResources := []map[string]interface{}{}
+
+	for _, uuid := range uuids.Fields {
+		resource, err := i.Document(string(uuid.Value()))
+		if err != nil {
+			log.Println("indexAndBackup#backup: ERR finding resource: ", string(uuid.Value()), err)
+			return
+		}
+
+		dd, err := MakeMapFromDocument(resource)
```
Comment (techieshark) [inline dba/index_and_backup.go]: Peter: "dd, what does that stand for? Dare Devil?" 

Senthil's response: "I didn't want to name it just 'd', so I called it 'dd'". 

FYI to future people: dd is a meta syntactic variable (like "foo" or "bar). But in this case, the fact that it begins with the fourth letter in the English language ('d') adds a bit of spice and should give the code reader the (correct) suspicion that this related to "document", and we might infer that the second 'd' is short for 'dictionary', which is actually a **d**ecent fit in this case! 

So in short, it's all good.

Comment (sent-hil) [inline dba/index_and_backup.go]: thanks

**Comment by techieshark** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/147)*

**Comment by techieshark** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/147#-:-KnCnxm3oP8dMVirvekf:bnfp4nl)*

**Pull request merged**